src/libpriv/rpmostree-util.cxx

```diff
@@ -589,11 +572,11 @@ rpmostree_deployment_get_layered_info (OstreeRepo        *repo,
   /* canonicalize outputs to empty array */
 
   if (out_is_layered != NULL)
-    *out_is_layered = is_layered;
+    *out_is_layered = layeredmeta.is_layered;
   if (out_layer_version != NULL)
-    *out_layer_version = clientlayer_version;
-  if (out_base_layer != NULL)
-    *out_base_layer = util::move_nullify (base_layer);
+    *out_layer_version = layeredmeta.clientlayer_version;
+  if (out_base_layer != NULL && layeredmeta.is_layered)
+    *out_base_layer = util::ruststr_dup_c_optempty(layeredmeta.base_commit);
```

Comment:
Oh I see, this does maintain that property. It's only in the new API that it changes. In that case, :heavy_check_mark: